### PR TITLE
Make text matchers reusable

### PR DESCRIPTION
**Pull request by provokateurin**
### What does it do?

Extracts the text matchers to a separate file.

### Why is it needed?

To make them reusable outside of the library, e.g. displaying message previews in a chat list and still get the matching.

### CODE REVIEW

**Comment by provokateurin** [comment]
@demchenkoalex Would you mind reviewing this?

**Comment by demchenkoalex** [comment]
Only can take a look on weekends, but also not every one of them, sometimes need to have time for myself. I see this PR so eventually it will land of course.

**Comment by provokateurin** [comment]
Take whatever time you need! Burnout is obviously not bad. Just wanted to know the status. I'm very grateful that this library exists :heart:

**Comment by provokateurin** [comment]
@demchenkoalex could you have a look at this again?

**Comment by provokateurin** [comment]
CI failure is fixed in https://github.com/flyerhq/flutter_chat_ui/pull/519

File: lib/src/widgets/message/system_message.dart [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
 
+import '../../../flutter_chat_ui.dart';
 import '../state/inherited_chat_theme.dart';
 
 /// A class that represents system message widget.
 class SystemMessage extends StatelessWidget {
   const SystemMessage({
-    super.key,
     required this.message,
+    this.options = const TextMessageOptions(),
```
Comment (demchenkoalex) [inline lib/src/widgets/message/system_message.dart:10]: this is not TextMessage, should have its own options

File: lib/src/widgets/message/system_message.dart [hunk 2]
```diff
@@ -1,25 +1,43 @@
 import 'package:flutter/material.dart';
 
+import '../../../flutter_chat_ui.dart';
```
Comment (demchenkoalex) [inline lib/src/widgets/message/system_message.dart:3]: Import looks strange, importing the whole chat UI for what? probably needs a specific file import

**Comment by demchenkoalex** [comment]
ok my comments will require larger refactor.. so just fixing the build is ok to merge

**Comment by provokateurin** [comment]
> so just fixing the build is ok to merge

What do you mean? I sent the build fix in another PR and now rebased this one on top of the merged PR.

**Comment by demchenkoalex** [comment]
> Can you elaborate? I'd love to get this refactored and merged :)

Basically you reused TextMessage parts in SystemMessage which is kinda a mess with import cycles. But it will be much larger refactor to move parts of TextMessage somewhere outside to then use in both TextMessage and SystemMessage so it is ok now, in any case, everything is re-done from scratch in v2

> What do you mean? I sent the build fix in another PR and now rebased this one on top of the merged PR.

Well you rebased just now, not when I wrote my comment :D Back then pipelines were failing still :)

**Comment by provokateurin** [comment]
Thanks a lot for merging :heart:

**Pull request merged**